src/inference_engines/vllm_engine.py

```diff
@@ -0,0 +1,152 @@
+import asyncio
+import json
+import os
+from io import IOBase
+from typing import BinaryIO, List, Union, get_args
+
+import torch
+from vllm import AsyncLLMEngine
+from vllm.engine.arg_utils import AsyncEngineArgs
+from vllm.sampling_params import SamplingParams
+
+FILE_LIKE = str | os.PathLike
+BYTES_LIKE = str | BinaryIO | IOBase | bytes
+
+
+class LoRA:
+
+    def __init__(self, adapter_config: Union[str, bytes, bytearray], adapter_model: FILE_LIKE) -> None:
+        self.adapter_config = json.loads(adapter_config)
+        self.adapter_model = torch.load(adapter_model, map_location="cpu")
+
+    @classmethod
+    def load_from_path(cls, adapter_config_path: os.PathLike, adapter_model_path: os.PathLike) -> "LoRA":
+        with open(adapter_config_path, "r") as f:
+            adapter_config = f.read()
+
+        with open(adapter_model_path, "rb") as f:
+            adapter_model = f.read()
+
+        return cls(adapter_config=adapter_config, adapter_model=adapter_model)
+
+    @classmethod
+    def load_from_bytes(cls, adapter_config_bytes: BYTES_LIKE, adapter_model_bytes: BYTES_LIKE) -> "LoRA":
+        return cls(adapter_config=adapter_config_bytes, adapter_model=adapter_model_bytes)
+
+
+# TODO (Moin): this class should inherit from engine
```

Comment:
Looks like you've already implemented most of the interface, so closing out this todo by making `vLLMEngine` inherit from the `Engine` base class should be pretty easy.